rust/cubesql/cubesql/src/compile/engine/udf/common.rs

```diff
@@ -2313,6 +2313,78 @@ macro_rules! generate_series_udtf {
+    ($ARGS:expr, $TYPE: ident, $PRIMITIVE_TYPE: ident) => {{
+        let mut section_sizes: Vec<usize> = Vec::new();
+        let l_arr = &$ARGS[0].as_any().downcast_ref::<PrimitiveArray<$TYPE>>();
+
+        if l_arr.is_some() {
+            let l_arr = l_arr.unwrap();
+            let r_arr = downcast_primitive_arg!($ARGS[1], "right", $TYPE);
+            let step_arr = IntervalMonthDayNanoArray::from_value(
+                IntervalMonthDayNanoType::make_value(0, 1, 0), // 1 day as default
+                1,
+            );
+            let step_arr = if $ARGS.len() > 2 {
+                downcast_primitive_arg!($ARGS[2], "step", IntervalMonthDayNanoType)
+            } else {
+                &step_arr
+            };
+
+            let mut builder = PrimitiveBuilder::<$TYPE>::new(1);
+            for (i, (start, end)) in l_arr.iter().zip(r_arr.iter()).enumerate() {
+                let step = if step_arr.len() > i {
+                    step_arr.value(i)
+                } else {
+                    step_arr.value(0)
+                };
+
+                if let (Some(start), Some(end)) = (start, end) {
+                    let mut section_size: i64 = 0;
+                    if start <= end && step > 0 {
+                        let mut current = start;
+                        loop {
+                            if current > end {
+                                break;
+                            }
+                            builder.append_value(current).unwrap();
+
+                            section_size += 1;
+
+                            if $ARGS[0].as_any().downcast_ref::<Date32Array>().is_some() {
+                                let current_dt =
+                                    NaiveDateTime::from_timestamp_opt((current as i64) * 86400, 0)
+                                        .unwrap();
+                                let res =
+                                    date_addsub_month_day_nano(current_dt, step, true).unwrap();
+                                current = (res.timestamp() / 86400) as $PRIMITIVE_TYPE;
+                            } else if $ARGS[0]
+                                .as_any()
+                                .downcast_ref::<TimestampNanosecondArray>()
+                                .is_some()
+                            {
+                                let current_dt = NaiveDateTime::from_timestamp_opt(
+                                    (current as i64) / 1_000_000_000,
+                                    (current % 1_000_000_000) as u32,
+                                )
+                                .unwrap();
+                                let res =
+                                    date_addsub_month_day_nano(current_dt, step, true).unwrap();
+                                current = res.timestamp_nanos_opt().unwrap() as $PRIMITIVE_TYPE;
+                            } else {
+                                panic!("Unsupported type");
```

Comment:
When something can fail, it will fail.

1. When the error depends on data (input, arguments, options). it's important to write type of it.
2. When you can return error, then you should return error. Please replace panic with error.

Ty

Reply:
Good catch! Sorry, there was error here of course, missed during debugging %)